Rock/Security/Authentication/Google.cs

```diff
@@ -0,0 +1,363 @@
+                        username = GetGoogleUser(googleUser, accessToken);
+                    }
+                }
+            }
+
+            catch ( Exception ex )
+            {
+                ExceptionLogService.LogException(ex, HttpContext.Current);
+            }
+
+            return !string.IsNullOrWhiteSpace(username);
+        }
+
+        /// <summary>
+        /// Gets the URL of an image that should be displayed.
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override String ImageUrl( )
+        {
+            return ""; /*~/Assets/Images/facebook-login.png*/
+        }
+
+        private string GetRedirectUrl( HttpRequest request )
+        {
+            Uri uri = new Uri(request.Url.ToString());
+            return uri.Scheme + "://" + uri.GetComponents(UriComponents.HostAndPort, UriFormat.UriEscaped) + uri.LocalPath;
+        }
+
+        /// <summary>
+        /// Authenticates the user based on user name and password
+        /// </summary>
+        /// <param name="user">The user.</param>
+        /// <param name="password">The password.</param>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override bool Authenticate( UserLogin user, string password )
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Encodes the password.
+        /// </summary>
+        /// <param name="user">The user.</param>
+        /// <param name="password">The password.</param>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override string EncodePassword( UserLogin user, string password )
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether [supports change password].
+        /// </summary>
+        /// <value>
+        /// <c>true</c> if [supports change password]; otherwise, <c>false</c>.
+        /// </value>
+        public override bool SupportsChangePassword
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Changes the password.
+        /// </summary>
+        /// <param name="user">The user.</param>
+        /// <param name="oldPassword">The old password.</param>
+        /// <param name="newPassword">The new password.</param>
+        /// <param name="warningMessage">The warning message.</param>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override bool ChangePassword( UserLogin user, string oldPassword, string newPassword, out string warningMessage )
+        {
+            warningMessage = "not supported";
```

Comment:
This looks a little sketch, not to mention the user experience if they actually tried it.  I believe there is a Password Change Not Supported clause in the Active Directory provider that you'd want to model it after.